### PR TITLE
[4.x] Split global set variables into its own repository and stache store

### PR DESCRIPTION
**Pull request by ryanmitchell**
There seems to be a lot of people who want to split config from data when using eloquent driver, which is already mostly possible apart from forms and globals. Forms will be possible if/when [this merges](https://github.com/statamic/cms/pull/6455).

See for reference: 
https://github.com/statamic/eloquent-driver/issues/172
https://github.com/statamic/eloquent-driver/issues/128
https://github.com/statamic/eloquent-driver/issues/147
https://github.com/statamic/eloquent-driver/pull/168

This PR splits global variables out into its own repository and store but maintains sharing the same directory as globals, so at a file-level nothing changes. All tests are passing locally which is a good sign that I haven't broken anything, and I was trying not to affect the existing structure in any way so made changes around it rather than simply adding a new directory for variables.

Could you let me know if this is something you would consider merging, and if so I'll spend some more time ensure the test coverage is up to scratch.

[Also, sorry. I know my PRs are always time consuming to review]

### CODE REVIEW

**Comment by jasonvarga** [comment]
> Also, sorry. I know my PRs are always time consuming to review

The best ones are.

**Comment by fdeneux** [comment]
Thanks @ryanmitchell , our team is missing this feature. @jasonvarga any idea when your team will be able to review (and consider merging) this PR?

We work with multiple environments so we like to split config from data, with config being version controlled and data environment specific. We've managed to so for all stores except globals, meaning that our team is required to replicate config changes for globals on all environments.

File: src/Stache/Stores/GlobalVariablesStore.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            parent::save($variable);
+        }
+    }
+
+    public function delete($variable)
+    {
+        if (Site::hasMultiple()) {
+            parent::delete($variable);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (ryanmitchell) [reply src/Stache/Stores/GlobalVariablesStore.php:84]: I remember commenting that out a lot to try and get tests to pass. I always felt like it should always save, just couldn't figure out why it caused test failures.

File: src/Stache/Stores/GlobalVariablesStore.php
Comment (ryanmitchell) [inline src/Stache/Stores/GlobalVariablesStore.php:1]: Now that I see it, its so obvious.
🤷

**Comment by ryanmitchell** [comment]
Thank you. Did it end up helping fix the other issue (#8505)?

**Comment by jasonvarga** [comment]
Yep it did, and it's what I focused most on with this PR.

But right after merging this I realized this PR totally doesn't work when you have a single site and with the variables directory different from the globals. The variables store doesn't find the base file. Kinda had my multi-site blinders on while reviewing this.

I'm thinking that a simple temporary solution is to throw an exception when you're using the stache and have different directories defined for globals and global-variables.

**Comment by ryanmitchell** [comment]
Yeah that’s a good point. Never thought of trying different directories as the files are merged in that situation. 

I think that’s a reasonable solution. Maybe in statamic 5 you always split the files even in single site mode?

**Pull request merged**